mongo/client_examples_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func ExampleConnect_replicaSet() {
+	// Create and connect a Client to a replica set deployment.
+	// Given this URI, the Go driver will first communicate with localhost:27017 and use the response to discover
+	// any other nodes in the replica set.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?replicaSet=replset")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Call Ping to verify that the deployment is up and the Client was configured successfully.
```

Comment:
Use of Ping only here could imply that it's only for replica sets.  Maybe make a `ExampleConnect_verify` example with it instead? Or use Ping everywhere.  Also probably worth adding to the comment that use of Ping reduces application resilience because applications starting up (e.g. autoscaling due to a load spike) will error if the database is temporarily unavailable or in the process of failing over (e.g. while scaling up to larger hardware due to the load spike).

Reply:
Moved the `Ping` call to a new example (`ExampleConnect_ping`) and added a note about application resiliency. I also updated the `Ping` documentation in https://github.com/mongodb/mongo-go-driver/pull/212 to talk about resiliency.